doc/guide.md

```diff
@@ -1937,6 +1938,31 @@ descriptor. For the fully up-to-date definition, you should always
 refer to the RAML and JSON schemas in the [Reference](#web-service)
 section.
 
+### Permissions and the `_tenantPermissions` Interface
+As desribed earlier, permissions and permissionSets can be defined in the ModuleDescriptor, but how does mod-permissions learn about them?  The answer is via another system interface: `_tenantPermissions`.  This interface, implemented by mod-permissions, is invoked by OKAPI when a module is enabled for a tenant.  The permissions defined in that module's ModuleDescriptor are furnished, allowing the permissions module to perform the necessary CRUD operations against permission storage.
+
+When mod-permissions is enabled for a tenant, OKAPI will invoke the `_tenantPermissions` API for all modules currently enabled for that tenant.  This is needed to inform the permissions module of any permissions defined in module descriptors enabled for the tenant prior to mod-permissions.
+
+Note that OKAPI is only responsible for providing the permissions and the corresponding moduleId to the `_tenantPermissions` API.  Determination of the appropriate actions to be taken is the responsibility of the permissions module.   Most of this will happen without explicit or special provisioning in the ModuleDescriptor.  Adding, removing or updating the permissions in the `permissionsSet` proprety is sufficient.  One exception to that is renaming of permissions.   If you wish to rename an existing permission, use the `renamedFrom` property on the permission object, e.g. 
+```
+...
+  "permissionSets": [
+    {
+      "permissionName": "test-basic.collection.get",
+      "displayName": "test-basic list records",
+      "description": "Get a list of records",
+	  "renamedFrom": [ "test-basic.get.list", "test-basic.view.list" ]
+    }, 
+...
+```
+Assuming `test-basic.get.list` and `test-basic-view.list` were defined in an versions of the ModuleDescriptor, this would result in:
```

Comment:
```suggestion
Assuming `test-basic.get.list` and `test-basic-view.list` were defined in an earlier version of the ModuleDescriptor, this would result in:
```

Reply:
used previous as suggested in another comment.